File: C/branch_prediction.c
#include <stdio.h>
#include <time.h>

int main() {
    int arr[1000000], sum = 0;
    for (int i = 0; i < 1000000; i++) arr[i] = i % 2; // Unpredictable pattern

    clock_t start = clock();
    for (int i = 0; i < 1000000; i++) {
        if (arr[i] == 0) sum++;
    }
    clock_t end = clock();

    printf("Execution Time: %lf seconds\n", (double)(end - start) / CLOCKS_PER_SEC);
    return 0;
}

File: PHP/branch_prediction.php
<?php
$arr = array_fill(0, 1000000, 0); // Predictable sequence
$sum = 0;

$start = microtime(true);
foreach ($arr as $val) {
    if ($val == 0) {
        $sum++;
    }
}
$end = microtime(true);

echo "Execution Time: " . ($end - $start) . " sec\n";
?>
